pkg/utils/k8sutil.go

```diff
@@ -233,13 +235,29 @@ func GetFunctionData(runtime, ftype, modName string) (imageName, depName, fileNa
 		}
 		if ftype == pubsubFunc {
 			if rVersion.pubsubImage == "" {
-				err = errors.New("The given runtime and version does not have a valid image for event based functions")
+				var runtimeList []string
+				for i := range runtimeObjList {
+					for j := range runtimeObjList[i] {
+						if runtimeObjList[i][j].pubsubImage != "" {
+							runtimeList = append(runtimeList, runtimeObjList[i][j].runtimeID+runtimeObjList[i][j].version)
+						}
+					}
+				}
+				err = errors.New("The given runtime and version '" + runtime + "does not have a valid image for event based functions. Available runtimes are: " + strings.Join(runtimeList[:], ", "))
```

Comment:
You are doing some for loops to list the available runtimes. Should it just be simple as a slice variable at top then we update it once there is new runtime supported ? Or if you want to keep it, should it be a separate function ? as there is duplicated code at the `rVersion.httpImage == ""` case right below